src/test/java/org/opensearch/security/support/HostAndCidrMatcherTest.java

```diff
@@ -0,0 +1,225 @@
+
+    @Test
+    public void shouldMatchIpv6WithinCidrRange() throws Exception {
+        matcher = new HostAndCidrMatcher(Arrays.asList(IPV6_DOCUMENTATION_CIDR));
+        InetAddress address = InetAddress.getByName(IPV6_DOCUMENTATION_IP);
+        assertThat(matcher.matchesCidr(address), is(true));
+    }
+
+    @Test
+    public void shouldHandleNullAddressInCidrMatching() {
+        matcher = new HostAndCidrMatcher(Arrays.asList(PRIVATE_CLASS_C_CIDR));
+        assertThat(matcher.matchesCidr(null), is(false));
+    }
+
+    @Test
+    public void shouldMatchValidHostnamePattern() throws Exception {
+        matcher = new HostAndCidrMatcher(Arrays.asList(EXAMPLE_DOMAIN, OPENSEARCH_WWW));
+        InetAddress address = InetAddress.getByName(EXAMPLE_WWW);
+        assertThat(matcher.matchesHostname(address, "ip-hostname"), is(true));
+    }
+
+    @Test
+    public void shouldHandleNullAddressInHostnameMatching() {
+        matcher = new HostAndCidrMatcher(Arrays.asList(EXAMPLE_DOMAIN));
+        assertThat(matcher.matchesHostname(null, "ip-hostname"), is(false));
+    }
+
+    @Test
+    public void shouldMatchWhenIpMatchesCidrOnly() throws Exception {
+        matcher = new HostAndCidrMatcher(Arrays.asList(EXAMPLE_DOMAIN, PRIVATE_CLASS_C_CIDR));
+        InetAddress address = InetAddress.getByName(PRIVATE_CLASS_C_IP);
+        assertThat(matcher.matches(address, "ip-hostname"), is(true));
+    }
+
+    @Test
+    public void shouldMatchWhenHostnameMatchesOnly() throws Exception {
+        matcher = new HostAndCidrMatcher(Arrays.asList(EXAMPLE_DOMAIN, PRIVATE_CLASS_C_CIDR));
+        InetAddress address = InetAddress.getByName(EXAMPLE_WWW);
+        assertThat(matcher.matches(address, "ip-hostname"), is(true));
+    }
+
+    @Test
+    public void shouldNotMatchWhenNeitherMatches() throws Exception {
+        matcher = new HostAndCidrMatcher(Arrays.asList(EXAMPLE_DOMAIN, PRIVATE_CLASS_C_CIDR));
+        InetAddress address = InetAddress.getByName(OPENSEARCH_WWW);
+        assertThat(matcher.matches(address, "ip-hostname"), is(false));
+    }
+
+    @Test
+    public void shouldHandleEmptyPatternList() throws Exception {
+        matcher = new HostAndCidrMatcher(Collections.emptyList());
+        InetAddress address = InetAddress.getByName(PRIVATE_CLASS_C_IP);
+        assertThat(matcher.matches(address, "ip-hostname"), is(false));
+    }
+
+    @Test
+    public void shouldHandleInvalidCidrNotation() throws Exception {
+        matcher = new HostAndCidrMatcher(Arrays.asList("invalid/cidr/notation"));
+        InetAddress address = InetAddress.getByName(PRIVATE_CLASS_C_IP);
+        assertThat(matcher.matchesCidr(address), is(false));
+    }
+
+    @Test(expected = Exception.class)
+    public void shouldHandleMalformedIpAddresses() throws Exception {
+        matcher = new HostAndCidrMatcher(Arrays.asList(PRIVATE_CLASS_C_CIDR));
+        InetAddress address = InetAddress.getByName("invalid.ip.address");
+        matcher.matchesCidr(address);
```

Comment:
What line is actually throwing the exception in the method below? My guess would be InetAddress.getByName(). If that's the case, then this test can be omitted, as it does not really verify behavior of the matcher itself.